pkg/sshd/circuit.go

```diff
@@ -0,0 +1,56 @@
+package sshd
+
+import (
+	"fmt"
+	"sync/atomic"
+)
+
+type CircuitState uint32
+
+const (
+	OpenState   CircuitState = 0
+	ClosedState CircuitState = 1
+)
+
+func (c CircuitState) String() string {
+	if c == OpenState {
+		return "OPEN"
+	} else if c == ClosedState {
+		return "CLOSED"
+	} else {
+		return fmt.Sprintf("Unknown (%d)", c.toUint32())
+	}
+}
+
+func (s CircuitState) toUint32() uint32 {
+	return uint32(s)
+}
+
+// Circuit is a concurrency-safe data structure that can take one of two states at any point in time:
+//
+// - OpenState - non functional
+// - ClosedState - functional
+//
+//
+type Circuit struct {
+	bit uint32
+}
+
```

Comment:
basically, I'm using `atomic` here because the circuit can "get away" with simple bit flipping - and using atomic operations can easily do that in a cheaper way than with a mutex